### ISSUE TITLE
Bug with RBAC - custom roles. Users with "has same role as creator" don't see content in ListView.

### ISSUE THREAD

**Issue by Uzair-Fasih** [strapi/strapi#10010]
### Describe the bug

When using a custom role with the defined condition of "**Has same role as creator**", the users with said role are unable to view the content in **ListView** which was created by another user with the same role. The entries created by the other users do show up, but all the custom fields other than the auto generated ones (created_at, published_at, etc) are populated with **'-'** instead of the actual data. Clicking on the entries to modify, however does show all the fields.

### Steps to reproduce the behavior

1. Create a brand new project with `yarn create strapi-app my-project --quickstart`
2. Add the EE license in to `.env` or `license.txt` (I used a trial license)
3. Rebuild the admin panel with `yarn build --clean`
4. Start the project with `yarn develop`
5. Create a sample model (called `article` perhaps) with a couple of fields (for eg: `title`, `content`, `isActive`)
**title**: string
**body**: richtext
**isActive**: Boolean
6. Create a new role `test`
7. Assign all field permissions but apply custom conditions
        **Can Create**: Anytime
        **Can Read**: Is creator and Has same role as creator
        **Can Update**: Is creator
        **Can Delete**: Is creator
8. Create 2 test users (User One and User two)
9. Create an article on one
10. Attempt to read on the other
### Expected behavior

Should be able to view all the content in the ListView

### Screenshots
- User One creates the article; Can view it
![image](https://user-images.githubusercontent.com/41604257/114301219-85521400-9ae1-11eb-8746-20f63047b2a7.png)
- User Two attempts to read the article; Cannot view content. (Although the entry for it remains visible)
![image](https://user-images.githubusercontent.com/41604257/114301247-b0d4fe80-9ae1-11eb-9ef6-91bc88ad9467.png)

### System

- Node.js version: v14.15.4
- NPM version: 6.14.10
- Strapi version: 3.5.4
- Database: SQLite (but should be reproducable with others)
- Operating system: Windows 10 with WSL

### Additional context
[#8129](https://github.com/strapi/strapi/issues/8129)
Similar instance of the problem

**Comment by derrickmehaffy** [comment]
@Uzair-Fasih I can't reproduce this on the same version, if possible can you push your test project into a git remove (just remember to remove your trial license, I can generate a test license to validate).

Or can you zip up (compress) and pass it to me either on our slack/forum via a PM?

**Comment by Uzair-Fasih** [comment]
@derrickmehaffy I followed the steps outlined above in the issue and was able to reproduce the bug again. I uploaded my project files excluding the `.cache` and `node_module` folders [here](https://github.com/Uzair-Fasih/rbac-bug). I have included the SQLite DB in the repo and the test user credentials are in the readme. Should be easy to observe by just cloning the project, installing dependencies and license and running the project.

**Comment by mickeeyo** [comment]
I'm experiecing same issue (version 3.6.3). I think It's happening because `roles` are missing in `created_by` object.
`findOne` controller is calling method `findOneWithCreatorRoles` which calls method `assocCreatorRoles` which do the job.  
Imho `findWithRelationCounts` controller should do something simillar. I've also tried to pass 'created_by.roles' as populate parameter but without success.
I was able to fix it by adding this line after getting result and pagination (in controllers/collection-types.js)
`const resultWithRoles = await Promise.all(results.map(async(entity) => {return await entityManager.assocCreatorRoles(entity)}));`
